### DISCUSSION TITLE
Opinion: Using Firebase functions for hosting NEXT.js is a very bad idea

### DISCUSSION THREAD

**Discussion by GorvGoyl** [vercel/next.js#11848] [Help]
I have recently started learning next.js and integrating it with Firebase. So, I might be wrong but this is my take:

Suppose we have next.js website which DOES NOT use `getServerSideProps` and `getInitialProps`.
We now configure Firebase to serve our next.js website from Firebase cloud functions instead of firebase hosting and suffer from these:
- we lose all CDN page caching benefits because now pages are built at server during run time by firebase functions and then sent to user i.e. when user visits website.  (PLEASE correct me if I'm wrong)
- firebase functions are slow as compared to firebase hosting and also suffers from cold start when not enough active users. i.e. first visit will takes time, sometimes 5 seconds.
- I'm not sure if it also affects SEO?

Almost all articles on the web including next.js and firebase are about setting up firebase functions to host next.js website. I just don't get it! Why can't we use firebase hosting and to show personal data (user profile,dashboard) show a static page first and then fetch the personal data from client side using rest api.

Examples from next.js repo suggesting to use firebase functions for hosting next.js:
- https://github.com/zeit/next.js/tree/master/examples/with-firebase-hosting
- https://github.com/zeit/next.js/tree/master/examples/with-firebase-hosting-and-typescript
on a seperate note, it's way too complicated to setup firebase like this. Even the official repos are outdated and doesn't work for latest version.

**Comment by david0178418** [comment]
> we lose all CDN page caching benefits because now pages are built at server during run time by firebase functions and then sent to user i.e. when user visits website. (PLEASE correct me if I'm wrong)

I'm in the middle of investigating nextjs capabilities for potential future solutions, and stumbled across this.

Being unfamiliar with nextjs, I don't know much about the trickiness that might arise from the intermingling of it and firebase, but I do know that you can still take advantage of the cdn caching.

[See here for an example](https://www.youtube.com/watch?v=LOeioOKUKI8&t=4m40s)

Of course, if your page updates frequently, you'd want to use your judgement on how long the CDN cache should last, but it avoids the issue of your server getting potentially hammered, off shooting subsequent requests to the cdn assets.

Hope that helps!
